pkg/db/canary.go

```diff
@@ -69,19 +77,31 @@ func PersistCheck(check pkg.Check, canaryID uuid.UUID) (uuid.UUID, error) {
 	}
 	check.Name = name
 	check.Description = description
+
+	// TODO: Find root cause why pod check has these labels in check model
+	delete(check.Labels, "canary-checker.flanksource.com/podCheck")
+	delete(check.Labels, "canary-checker.flanksource.com/check")
```

Comment:
These labels are added by the pod check run, but they keep on changing leading to excessive upserts. Adding a TODO for the root cause currently